src/strategies/bmc/index.ts

```diff
@@ -0,0 +1,87 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { Multicaller } from '../../utils';
+
+export const author = 'justinrwong';
+export const version = '0.0.1';
+
+const ogAbi = [
+  'function walletMiners(address account) external view returns (uint256[])'
+];
+
+const ultraAbi = [
+  'function tokensOfOwner(address account) external view returns (uint256[])'
+];
+
+const stakingAbi = [
+  'function getStakedOGs(address account) external view returns (uint256[])',
+  'function getStakedULTRAs(address account) external view returns (uint256[])'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  // Setup multicall for the OG/Ultra/Staking Contracts
+  const ogToken = new Multicaller(network, provider, ogAbi, { blockTag });
+  const ultraToken = new Multicaller(network, provider, ultraAbi, { blockTag });
+  const stakingOGToken = new Multicaller(network, provider, stakingAbi, {
+    blockTag
+  });
+  const stakingUltraToken = new Multicaller(network, provider, stakingAbi, {
+    blockTag
+  });
+  addresses.forEach((address) => {
+    ogToken.call(address, options.og, 'walletMiners', [address]);
+    ultraToken.call(address, options.ultra, 'tokensOfOwner', [address]);
+    stakingOGToken.call(address, options.staking, 'getStakedOGs', [address]);
+    stakingUltraToken.call(address, options.staking, 'getStakedULTRAs', [
+      address
+    ]);
+  });
+
+  function _throw(m) {
+    throw m;
+  }
+
+  // when no assets are held, the contract throws an error, which means it's empty.
+  const [ogResponse, ultraResponse, stakedOgResponse, stakedUltraResponse]: [
+    Record<string, BigNumber[]>,
+    Record<string, BigNumber[]>,
+    Record<string, BigNumber[]>,
+    Record<string, BigNumber[]>
+  ] = await Promise.all([
+    ogToken
+      .execute()
+      .catch((error) => (error.code === 'CALL_EXCEPTION' ? [] : _throw(error))),
```

Comment:
Hello @JustinRWong in cases where the query hit gas cap of the node we are using, then also we may hit `CALL_EXCEPTION` better to check the message `Address owns no tokens` instead, we can't access this message.?

Reply:
No, we can't access the message because the `CALL_EXCEPTION` ends up coming from the `require` statement of the smart contract

Reply:
Example of how this could ignore other errors as well:
`staking` contract doesn't exist at the `snapshot` block given in examples.json and throws an error, and because of the catch  error is hidden and voting power will be `0` 

There can be a lot of exception errors. and voting power will be inconsistent

Reply:
I don't see a solution for this. maybe would be helpful if you have a subgraph for this contract https://etherscan.io/address/0x0c6822ca73de6871f27acd9ca05a05b99294b805#readContract